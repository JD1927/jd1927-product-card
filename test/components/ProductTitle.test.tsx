import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
  test('Should show the component successfully with the custom title', () => {
    // Arrange
    const wrapper = renderer.create(
      <ProductTitle title='Custom product' className='custom-class'/>
    );
    // Act
    // Assert
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('Should show the component with the product name', () => {
    // Arrange
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          () => (<ProductTitle/>)
        }
      </ProductCard>
    );
    // Act
    // Assert
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
