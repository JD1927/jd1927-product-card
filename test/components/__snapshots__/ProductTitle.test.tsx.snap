// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Should show the component successfully with the custom title 1`] = `
<span
  className="productDescription custom-class"
>
  Custom product
</span>
`;

exports[`ProductTitle Should show the component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without Image
  </span>
</div>
`;
